### PR TITLE
Feature/702 Prepare transaction payload for circuit input

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

This PR addresses 2 tasks from #702 

- Translate tx.payload and current bpi account state into appropriate circuit inputs
- Hash(root of the merkelized invoice + Hash (witness))

## Related Issue

#702 

## How Has This Been Tested

Manual test by running locally. Automated testing would be included in future PRs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts [hunk 1]
```diff
@@ -2,15 +2,24 @@ import { Injectable } from '@nestjs/common';
 import { Witness } from '../../../models/witness';
 import { Proof } from '../../../models/proof';
 import { ICircuitService } from '../circuitService.interface';
-import { computeEcdsaPublicInputs } from './utils/ecdsa/computeEcdsaPublicInputs';
+import {
+  computeEcdsaPublicInputs,
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts:6]:
```suggestion
  computeEcdsaSigPublicInputs,
```

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts [hunk 2]
```diff
@@ -70,31 +79,57 @@ export class SnarkjsCircuitService implements ICircuitService {
...
+    stateTree: MerkleTree;
+  }): Promise<object> {
+    //1. Ecdsa signature
     const { signature, Tx, Ty, Ux, Uy, publicKeyX, publicKeyY } =
-      computeEcdsaPublicInputs(
-        inputs['signature'],
-        inputs['messageHash'],
-        inputs['publicKey'],
-      );
+      computeEcdsaPublicInputs(inputs.tx);
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts:100]:
```suggestion
      computeEcdsaSigPublicInputs(inputs.tx);
```

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    Ux: circuitPubInput.Ux,
+    Uy: circuitPubInput.Uy,
+    publicKeyX: publicKeyCoordinates.getX().toString(),
+    publicKeyY: publicKeyCoordinates.getY().toString(),
+  };
+
+  return input;
+};
+
+export const computeEcdsaPublicInputs = (tx: Transaction) => {
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts:42]:
```suggestion
export const computeEcdsaSigPublicInputs = (tx: Transaction) => {
```

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+  const ecdsaSignature = ethers.utils.splitSignature(tx.signature);
+
+  const messageHash = ethers.utils.arrayify(
+    ethers.utils.hashMessage(tx.payload),
+  );
+
+  const publicKey = tx.fromBpiSubjectAccount.ownerBpiSubject.publicKey;
+
+  return computeEffectiveEcdsaPublicInputs(
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts:51]:
```suggestion
  return computeEffectiveEcdsaSigPublicInputs(
```

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
Left some suggestions @biscuitdey .. let's finalize the discussion on non-inclusion

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts (hunk 1 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts]: Fixed.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts (hunk 2 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/snarkjs.service.ts]: Fixed.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts (hunk 3 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts]: Fixed.

File: examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts (hunk 4 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/zeroKnowledgeProof/services/circuit/snarkjs/utils/computePublicInputs.ts]: Fixed.

**Review by ognjenkurtic — APPROVED** [review]

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Pull request merged**